Adding round robin option of source ip for UDPunreachable

kpetremann opened the pull request.

kpetremann committed:
Adding round robin option of source ip for UDPunreachable

fdomain commented on netprobify/protocol/target.py:
```diff
@@ -55,6 +58,9 @@ def __init__(
         src_ipv6 -- Bind probes to a specific source IPv6
+        src_subnet_ipv4 -- Source IPv4 subnet for src ip round robin
+        src_subnet_ipv6 -- Source IPv6 subnet for src ip round robin
+        src_ip_round_robin -- Enable/disable round robin on source IP address to help ECMP
```
`src_ip_round_robin` is not useful, since `src_subnet_ipv4` or `src_subnet_ipv6` are defined, source IP round robin should be implicity enabled. You can easily remove this arg and relies only on `src_subnet_ipv4/ipv6` arguments. It will simplify your code as you will have less checks to perform.

kpetremann replied:
removed

kpetremann committed:
Removing src_ip_round_robin

kpetremann committed:
Remove useless log

kpetremann committed:
edit comment

fdomain commented on netprobify/protocol/udp_unreachable.py:
```diff
@@ -136,10 +137,21 @@ def generate_packets(self, all_groups, id_gen, logging_level="WARNING"):
+            src_subnet = (
+                grp.src_subnet_ipv4 if self.address_family == "ipv4" else grp.src_subnet_ipv6
+            )
+            src_network = ip_network(src_subnet) if src_subnet else None
```
```suggestion
            src_network = ip_network(get_src_subnet(self, grp))
```
and add `get_src_subnet` function in your UDP class (or Target maybe as it could be reused for other probing type) doing the following : 
```
def get_src_subnet(self, grp):
  if self.address_family == "ipv4":
    src_subnet = grp.src_subnet_ipv4
  else:
    src_subnet = grp.src_subnet_ipv6
  return src_subnet if src_subnet else None
```

kpetremann replied:
you are right, but why not return ip_network object directly?

fdomain replied:
yes it would be even better

kpetremann replied:
done

kpetremann committed:
Combine IP and port round robin

priority: round robin on source IP
source port changes only when one cycle of source IP round robin is finished

it ensures that the ECMP is done properly: only one field changes for each packets

kpetremann committed:
Factorize get_src_subnet

fdomain reviewed (approved):
lgtm
